src/libstore/remote-store.cc

```diff
@@ -1084,6 +1084,18 @@ unsigned int RemoteStore::getProtocol()
     return conn->daemonVersion;
 }
 
+bool RemoteStore::checkTrustedUser()
+{
+    auto conn(getConnection());
+    if (GET_PROTOCOL_MINOR(conn->daemonVersion) >= 35) {
+        conn->to << wopCheckTrustedUser;
+        conn.processStderr();
+        return readInt(conn->from);
+    } else {
+      throw Error("daemon protocol %d is too old (< 2.13) to ask whether we're a trusted user",
+          GET_PROTOCOL_MINOR(conn->daemonVersion));
```

Comment:
Apparently you need to bit shift by 8 bytes to read `GET_PROTOCOL_MAJOR`. This was incredibly confusing to me because:
- `GET_PROTOCOL_MAJOR(conn->daemonVersion)` will return `256`
- `GET_PROTOCOL_MAJOR(conn->daemonVersion) >> 8` will return `1`

Anyway.. I've implemented it now as follows.

```diff
diff --git a/src/libstore/remote-store.cc b/src/libstore/remote-store.cc
index 154afa726..5cfd432dd 100644
--- a/src/libstore/remote-store.cc
+++ b/src/libstore/remote-store.cc
@@ -1092,7 +1092,8 @@ bool RemoteStore::checkTrustedUser()
         conn.processStderr();
         return readInt(conn->from);
     } else {
-      throw Error("daemon protocol %d is too old (< 2.13) to ask whether we're a trusted user",
+      throw Error("daemon protocol %d.%d is too old (< 2.13) to ask whether we're a trusted user",
+          GET_PROTOCOL_MAJOR(conn->daemonVersion) >> 8,
           GET_PROTOCOL_MINOR(conn->daemonVersion));
     }
 }
```

Reply:
https://github.com/NixOS/nix/pull/7523